operator/src/main/java/oracle/kubernetes/operator/Watcher.java

```diff
@@ -126,7 +126,13 @@ private void watchForEvents() {
                 .withResourceVersion(resourceVersion.toString())
                 .withTimeoutSeconds(tuning.watchLifetime))) {
       while (watch.hasNext()) {
-        Watch.Response<T> item = watch.next();
+        Watch.Response<T> item;
+        try {
+          item = watch.next();
+        } catch (Throwable e) {
+          watch.discardClient();
+          throw e;
```

Comment:
Is the socket exception still logged as a WARNING?   If so, can we (A) detect the exact exception is a socket exception, and (B) only log a FINE in taht case?

Reply:
Unfortunately, we can't.  These exceptions are getting printed by threads that the HTTP client creates.  I have some (not done) work where I override the ThreadFactory of the HTTP client library and attempt to suppress this and other exceptions.

Reply:
Worth exploring. Not sure yet how to distinguish, as we just get an exception from the k8s api

Reply:
Yes, but the exception has already been printed by another thread.